specs/trip_dispatch_spec.rb

```diff
@@ -111,4 +95,67 @@
       expect(passenger.trips).must_include trip
     end
   end
+
+  describe '#request_trip method' do
+    before do
+      @requested_trip = @dispatcher.request_trip(1)
+    end
+
+    it "generates an array of :AVAILABLE drivers" do
+      available_drivers = @dispatcher.generate_available_drivers
+      expect(available_drivers).must_be_kind_of Array
+      expect(available_drivers.first.status).must_equal :AVAILABLE
+      expect(available_drivers.last.status).must_equal :AVAILABLE
+    end
+
+    it "returns an array of :AVAILABLE drivers that aren't the Passenger" do
+      available_drivers = @dispatcher.check_drivers_not_passenger?(2)
+      expect(available_drivers).must_be_kind_of Array
+
+      available_drivers.each do |driver|
+        expect(driver.id).wont_equal 2
+      end
+    end
+
+    it "Assigns the first :AVAILABLE driver with no driven trips to a new Trip" do
+      driver  = @requested_trip.driver
+      expect(driver).must_be_instance_of RideShare::Driver
+      expect(driver.driven_trips.length).must_equal 1
+    end
+
+    it "Assigns the first :AVAILABLE driver with the oldest recent trip if no new drivers are available to a new Trip" do
+      @requested_trip = @dispatcher.request_trip(1)
+      driver  = @requested_trip.driver
+      expect(driver).must_be_instance_of RideShare::Driver
+      expect(driver.driven_trips.length).must_equal 4
+    end
+
+    it "uses the current time as @start_time for new Trip" do
+      start_time = @requested_trip.start_time
+      expect(start_time).must_be_instance_of Time
+    end
+
+    it "Adds new Trip to collection of all Trips in TripDispatcher" do
+      expect(@dispatcher.trips.last).must_equal @requested_trip
+    end
+
+    it "Does not assign a driver with same ID as passenger" do
+      new_trip = @dispatcher.request_trip(8)
+      driver_id = new_trip.driver.id
+      passenger_id = new_trip.passenger.id
+
+      expect(driver_id).wont_equal passenger_id
+      expect(driver_id).must_equal 5
```

Comment:
Would it normally assign driver 8 in this case? How do you know? How could you make it clear to the reader of this test?

A comment explicitly calling out this assumption would be a good start, but even better would be some assertions before your code that verify the expected behavior. For example, you might call `@dispatcher.first_available_driver` with a different passenger ID, and make sure that you get driver 8 back.